pageserver/src/tenant_mgr.rs

```diff
@@ -379,8 +345,11 @@ pub fn set_tenant_state(tenant_id: ZTenantId, new_state: TenantState) -> anyhow:
                 Some(tenant_id),
                 None,
             );
-            thread_mgr::shutdown_threads(Some(ThreadKind::GarbageCollector), Some(tenant_id), None);
-            thread_mgr::shutdown_threads(Some(ThreadKind::Compactor), Some(tenant_id), None);
+
+            // Wait until all gc/compaction tasks finish
+            // TODO send cancellation signal too, or make the state a watch
```

Comment:
This is necessary because otherwise we'll wait an entire gc_period. Will fix before merge

Reply:
Fixed the test instead. It's not a big problem